tests/components/template/test_alarm_control_panel.py

```diff
@@ -0,0 +1,556 @@
+                        "arm_home": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_night": {"service": "test.automation"},
+                        "arm_away": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "disarm": {
+                            "service": "alarm_control_panel.alarm_disarm",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    service_calls = async_mock_service(hass, "test", "automation")
+
+    await common.async_alarm_arm_night(
+        hass, entity_id="alarm_control_panel.test_template_panel"
+    )
+    await hass.async_block_till_done()
+
+    assert len(service_calls) == 1
+
+
+async def test_disarm_action(hass):
+    """Test disarm action."""
+    await setup.async_setup_component(
+        hass,
+        "alarm_control_panel",
+        {
+            "alarm_control_panel": {
+                "platform": "template",
+                "panels": {
+                    "test_template_panel": {
+                        "value_template": "{{ states('alarm_control_panel.test') }}",
+                        "arm_home": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "disarm": {"service": "test.automation"},
+                        "arm_away": {
+                            "service": "alarm_control_panel.alarm_arm_home",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                        "arm_night": {
+                            "service": "alarm_control_panel.alarm_disarm",
+                            "entity_id": "alarm_control_panel.test",
+                            "data": {"code": "1234"},
+                        },
+                    }
+                },
+            }
+        },
+    )
+
+    await hass.async_start()
+    await hass.async_block_till_done()
+
+    service_calls = async_mock_service(hass, "test", "automation")
+
+    await common.async_alarm_disarm(
+        hass, entity_id="alarm_control_panel.test_template_panel"
+    )
+    await hass.async_block_till_done()
+
+    assert len(service_calls) == 1
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4788050)